pkg/trainer/training.go

```diff
@@ -405,3 +405,7 @@ func (j *TrainingJob) name() string {
 func (j *TrainingJob) fullname() string {
 	return j.job.ObjectMeta.GetNamespace() + ":" + j.job.ObjectMeta.GetName()
 }
+
+func (j *TrainingJob) SchedulerName() string {
+	return j.job.Spec.SchedulerName
```

Comment:
Why do we need a special Getter for SchedulerName?
We can just get it directly form `j.job.Spec.SchedulerName`.

Reply:
Because `TrainingJob.job` is a private member so `TrainingJob` should provide the accessor method.

Reply:
Yes, the current structure is a little complex:
- We have `TFJob` for API object Spec/Status
- Then create `TrainingJob` and `TFReplicaSet` which keep many cache of TFJob

And PodSpec already have a field [SchedulerName](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2927), can we reuse it?

Reply:
`PodSpec.SchedulerName` isn't suitable for the purpose. See the discussion start with this message https://github.com/kubeflow/tf-operator/pull/398#issuecomment-368017331